src/transformers/util.ts

```diff
@@ -508,3 +508,16 @@ export function extractAttributeAsNumeric(
 
   return numericValues;
 }
+
+export function makeDatasetImmutable(dataset: DataSet): DataSet {
+  const newCollections = dataset.collections.map(cloneCollection);
+  for (const c of newCollections) {
+    if (c.attrs) {
+      c.attrs.map((attr) => (attr.editable = false));
+    }
+  }
+  return {
+    collections: newCollections,
+    records: dataset.records,
+  };
+}
```

Comment:
Nice, this is already a big improvement over the previous. I wonder, is there any way to disable dragging (or renaming) of attributes? Or is editable=false all we have for immutability (maybe this is something we could request from Bill and Jonathan--it might not be hard to do given #21). 

This would just prevent more cases in which the user accidentally mutates the table and then gets confusing behavior on update.